libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1001,6 +1001,7 @@ void CompilerUtils::storeStringData(bytesConstRef _data)
 unsigned CompilerUtils::loadFromMemoryHelper(Type const& _type, bool _fromCalldata, bool _padToWords)
 {
 	unsigned numBytes = _type.calldataEncodedSize(_padToWords);
+	solAssert(numBytes > 0, "");
```

Comment:
When can this be 0?

Reply:
In rare cases. Basically if it is zero, then it cannot be encoded. I think adding an assert right into the `calldataEncodedSize` function did not work in the past because it was called just to check.

The only cases where it can be zero are when it is a storage struct that contains a mapping or an array that contains such a thing.

Reply:
Can we have a test case which triggers this assert?

Reply:
If the assert is triggered, this means that the function was used incorrectly. We can create a unit test that uses the function incorrectly, but I don't really see the point...

Reply:
Do we have proper checks in the typechecker which can avoid failing this assert?

Reply:
This assert is the check. I'm not sure we are talking about the same thing...